boards/arm/frdm_kl25z/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+if(CONFIG_PINMUX_MCUX)
+  zephyr_library()
+  zephyr_library_include_directories(${PROJECT_SOURCE_DIR}/drivers)
+  zephyr_library_sources(pinmux.c)
+endif()
```

Comment:
what happens when we add additional sources here? those will not need to depend on PINUXMUx

Reply:
Depends on the source. If it is a simple source file with no special dependencies or build environment requirements then it can be simply

zephyr_sources(new_source.c)

If it has special requirements the user could create a named library for it.

zephyr_library_named(new_lib)
zephyr_library_change_env_somehow(foo)
zephyr_library_sources(new_source.c)